File: BingeBuddyNg/BingeBuddyNg.Spa/src/models/UserDTO.ts
import { VenueDTO } from 'src/models/VenueDTO';
import { UserInfoDTO } from './UserInfoDTO';
import { PushInfo } from './PushInfo';
import { FriendRequestDTO } from './FriendRequestDTO';
import { UserStatisticsDTO } from './UserStatisticsDTO';

export class UserDTO {
    id: string;
    name: string;
    profileImageUrl: string;
    weight?: number;
    pushInfo?: PushInfo;
    friends?: UserInfoDTO[];
    currentVenue?: VenueDTO;
    language?: string;
    currentStats: UserStatisticsDTO;
    incomingFriendRequests?: FriendRequestDTO[];
    outgoingFriendRequests?: FriendRequestDTO[];
}

File: BingeBuddyNg/BingeBuddyNg.Services/User/DTO/UserDTO.cs
﻿using System;
using System.Collections.Generic;
using BingeBuddyNg.Core.Infrastructure;
using BingeBuddyNg.Core.Statistics.DTO;
using BingeBuddyNg.Core.User.Domain;
using BingeBuddyNg.Core.Venue.DTO;

namespace BingeBuddyNg.Core.User.DTO
{
    public class UserDTO
    {
        public string Id { get; set; }

        public string Name { get; set; }

        public int? Weight { get; set; }

        public Gender Gender { get; set; }

        public string ProfileImageUrl { get; set; }

        public PushInfo PushInfo { get; set; }

        public List<UserInfoDTO> Friends { get; set; }

        public VenueDTO CurrentVenue { get; set; }

        public string Language { get; set; }

        public UserStatisticsDTO CurrentStats { get; internal set; }

        public DateTime LastOnline { get; set; }

        public List<FriendRequestDTO> IncomingFriendRequests { get; set; }

        public List<FriendRequestDTO> OutgoingFriendRequests { get; set; }
    }
}